src/storage/src/region/writer.rs

```diff
@@ -539,15 +548,10 @@ impl WriterInner {
                     }
                 }
 
-                if let Some(payload) = payload {
-                    num_requests += 1;
-                    // Note that memtables of `Version` may be updated during replay.
-                    let version = version_control.current();
-
-                    if req_sequence > last_sequence {
-                        last_sequence = req_sequence;
-                    } else {
-                        logging::error!(
+                if req_sequence > last_sequence {
```

Comment:
Could we add a test for this case? You can write a test similar to test_read_after_flush but just flush and restart.

https://github.com/GreptimeTeam/greptimedb/blob/8764ce78458c217a5a4cfc22bdb6f71feeed425f/src/storage/src/region/tests/flush.rs#L184